Assets/Project/Modules/Enemies/EnemyVFX/Scripts/EnemyVisuals.cs

```diff
@@ -36,23 +36,39 @@ private void Awake()
             }
         }
 
-        public void Configure()
+        public void Configure(IParticleFactory particleFactory)
         {
+            _particleFactory = particleFactory;
             _originalMeshDatas[0]._mesh.material.SetFloat("_Health", 1.0f);
         }
 
-        public virtual void PlayHitEffects(float healthCoef01)
+        public virtual void PlayHitEffects(float healthCoef01, DamageHit damageHit)
         {
             _originalMeshDatas[0]._mesh.material.SetFloat("_Health", healthCoef01);
-            
-            //smallest camera shake
 
+            ParticlesHitEffect(damageHit);
+            FlashHitEffect().Forget();
+        }
 
-            CircleEffect().Forget();
-            FlashEffect().Forget();
+        public virtual void PlayDeathEffects(DamageHit damageHit)
+        {
+            ParticlesHitEffect(damageHit);
+        }
+
+        private void ParticlesHitEffect(DamageHit damageHit)
```

Comment:
I guess this is "TODO" because you are waiting for me to add the functionalities on my end?

Reply:
Yes, it is only ment for easier access later on, so I do not need to figure out where exactly should the new code go in case I forget.